### PR TITLE
Added support for HTTP Headers

### PR DESCRIPTION
**Pull request by salah-ghanim**

### CODE REVIEW

File: EventSource/EventSource.h [hunk 1]
```diff
@@ -53,6 +53,19 @@ typedef void (^EventSourceEventHandler)(Event *event);
 /// @param timeoutInterval The request timeout interval in seconds. See <tt>NSURLRequest</tt> for more details. Default: 5 minutes.
 + (instancetype)eventSourceWithURL:(NSURL *)URL timeoutInterval:(NSTimeInterval)timeoutInterval;
 
+/// Returns a new instance of EventSource with the specified URL.
+///
+/// @param URL The URL of the EventSource.
+/// @param headers the HTTP Headers used in the request
++ (instancetype)eventSourceWithURL:(NSURL *)URL HTTPHeaders:(NSDictionary<NSString*, NSString *>*) headers;
```
Comment (neilco) [inline EventSource/EventSource.h:60]: If the `HTTPHeaders` can be `nil`, then add the `nullable` attribute to the parameter. This applies to all the new method signatures in this pull request.

Comment (neilco) [reply EventSource/EventSource.h:60]: Also, style: `NSDictionary<NSString*, NSString *>*` should be `NSDictionary<NSString *, NSString *> *`. This applies to all the new method signatures in this pull request.

File: EventSource/EventSource.m [hunk 2]
```diff
@@ -53,23 +54,43 @@ + (instancetype)eventSourceWithURL:(NSURL *)URL timeoutInterval:(NSTimeInterval)
...
         _listeners = [NSMutableDictionary dictionary];
         _eventURL = URL;
         _timeoutInterval = timeoutInterval;
         _retryInterval = ES_RETRY_INTERVAL;
-
+        _HTTPRequestHeaders = headers;
         messageQueue = dispatch_queue_create("co.cwbrn.eventsource-queue", DISPATCH_QUEUE_SERIAL);
         connectionQueue = dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_HIGH, 0);
-
+
```
Comment (neilco) [inline EventSource/EventSource.m:93]: Style: Removing the trailing whitespace on this otherwise empty line.

File: EventSource/EventSource.m [hunk 3]
```diff
@@ -53,23 +54,43 @@ + (instancetype)eventSourceWithURL:(NSURL *)URL timeoutInterval:(NSTimeInterval)
...
+- (instancetype) initWithURL:(NSURL *)URL timeoutInterval:(NSTimeInterval)timeoutInterval HTTPHeaders:(NSDictionary<NSString*, NSString*> *)headers
+{
     self = [super init];
     if (self) {
         _listeners = [NSMutableDictionary dictionary];
         _eventURL = URL;
         _timeoutInterval = timeoutInterval;
         _retryInterval = ES_RETRY_INTERVAL;
-
```
Comment (neilco) [inline EventSource/EventSource.m:69]: Style: Group together the setting of ivars.

File: EventSource/EventSource.m [hunk 4]
```diff
@@ -218,6 +239,12 @@ - (void)_open
     NSMutableURLRequest *request = [NSMutableURLRequest requestWithURL:self.eventURL
                                                            cachePolicy:NSURLRequestReloadIgnoringCacheData
                                                        timeoutInterval:self.timeoutInterval];
+    if (self.HTTPRequestHeaders) {
+        for (NSString * key in self.HTTPRequestHeaders.allKeys){
```
Comment (neilco) [inline EventSource/EventSource.m:243]: Style: The preferred style is:

```objc
for (NSString *headerField in self.HTTPRequestHeaders) {
    NSString *value = self.HTTPRequestHeaders[headerField];
    [request setValue:value forHTTPHeaderField:headerField];
}
```

**Review by neilco — CHANGES_REQUESTED** [review]

File: EventSource/EventSource.h (hunk 1 above)
Comment (salah-ghanim) [reply EventSource/EventSource.h:60]: @neilco introducing nullability to the header will introduce warning unless all properties and parameters have nullability attributes

**Comment by MorganLindqvist** [comment]
This looks like a nice addition, I would like to use it to enable authentication (cookies) for the SSE. Could the outstanding issues be fixed and the pull request landed shortly?

**Pull request closed without merging**